lib/fizzy/execute.cpp

```diff
@@ -159,6 +159,22 @@ Instance instantiate(const Module& module)
     // NOTE: fill it with zeroes
     bytes memory(memory_min * page_size, 0);
 
+    // Fill out memory based on data segments
+    for (const auto& data : module.datasec)
+    {
+        uint64_t offset;
+        if (data.offset.kind == ConstantExpression::Kind::Constant)
```

Comment:
How many cases are going to be here? Can you rewrite it in a way `uint64_t offset` is initialized when declared.

Reply:
Two cases. Constant or global and both evaluate to a single offset.